src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -0,0 +1,219 @@
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group BGE
+* @group-content //todo
+* @description Lightning Component Server Controller for the Data Import Batch Entry component.
+*/
+
+public with sharing class BGE_DataImportBatchEntry_CTRL {
+
+    /*******************************************************************************************************
+    * @description deletes DataImport__c record
+    * @param batchId: ID of the NPSP_Data_Import_Batch__c
+    * @param dataImportId: ID of the DataImport__c
+    * @return serialized string of DataImportRow for use in the lightning:datatable
+    */
+    @AuraEnabled
+    public static String deleteDataImportRow(Id batchId, Id dataImportId) {
+        delete new DataImport__c(Id = dataImportId);
+        return JSON.serialize(getDataImports(batchId));
+    }
+
+    /*******************************************************************************************************
+    * @description initializes the data model for the BGE_DataImportBatchEntry component
+    * @param batchId: ID of the NPSP_Data_Import_Batch__c
+    * @return serialized string of dataImportModel
+    */
+    @AuraEnabled
+    public static String getDataImportModel(Id batchId){
+
+        DataImportModel dataImportModel = new DataImportModel();
+
+        dataImportModel.id = batchId;
+        dataImportModel.dataImportRows = getDataImports(batchId);
+        dataImportModel.columns = getColumns();
+        dataImportModel.labels = getBatchDataImportLabels();
+        dataImportModel.isNamespaced = String.isNotBlank(UTIL_Namespace.getNamespace());
+
+
+        return JSON.serialize(dataImportModel);
+    }
+
+    @AuraEnabled
+    public static void updateDataImports(List<DataImport__c> diList) {
+        database.update(diList);
+    }
+
+    /*******************************************************************************************************
+    * @description queries the DataImport__c records associated with the given batchId
+    * @param batchId: ID of the NPSP_Data_Import_Batch__c
+    * @return list of custom DataImportRow objects for use in the lightning:datatable
+    */
+    @AuraEnabled
+    public static List<DataImportRow> getDataImports(Id batchId){
+
+        try {
+            List<DataImportRow> result = new List<DataImportRow>();
+            for (DataImport__c dataImportRecord :[SELECT Id
```

Comment:
I'd avoid included the query in a for loop when it's this involved. How about: 

```
List<DataImport__c> dataImportRecords = [
    SELECT Id      
        Name, 
        Contact1_Firstname__c, 
        Contact1_Lastname__c, 
        Contact1Imported__c, 
        Contact1Imported__r.FirstName, 
        Contact1Imported__r.LastName, 
        Contact1Imported__r.Name, 
        Account1Imported__c, 
        Account1Imported__r.Name, 
        Donation_Date__c, 
        Donation_Amount__c, 
        Donation_Donor__c, 
        Payment_Method__c, 
        Payment_Check_Reference_Number__c, 
    FROM DataImport__c
    WHERE NPSP_Data_Import_Batch__c = :batchId
    ORDER BY Id ASC
];

for (DataImport__c dataImportRecord : dataImportRecords) {
    result.add(new DataImportRow(dataImportRecord));
}
```

Reply:
This is fixed in the dryRun branch: https://github.com/SalesforceFoundation/Cumulus/compare/feature/BGE-LC__demo...feature/BGE-LC__dryrun